burn-book/src/building-blocks/module.md

```diff
@@ -99,3 +97,33 @@ pub trait ModuleMapper<B: Backend> {
     fn map<const D: usize>(&mut self, id: &ParamId, tensor: Tensor<B, D>) -> Tensor<B, D>;
 }
 ```
+
+## Built-in Modules
+
+Burn comes with a few built-in modules that you can use to build your own modules.
+
+| Burn API             | PyTorch Equivalent                      |
+| -------------------- | --------------------------------------- |
+| `AdaptiveAvgPool1d`  | `nn.AdaptiveAvgPool1d`                  |
+| `AdaptiveAvgPool2d`  | `nn.AdaptiveAvgPool2d`                  |
+| `AvgPool1d`          | `nn.AvgPool1d`                          |
+| `AvgPool2d`          | `nn.AvgPool2d`                          |
+| `BatchNorm`          | `nn.BatchNorm1d`, `nn.BatchNorm2d` etc. |
+| `Conv1d`             | `nn.Conv1d`                             |
+| `Conv2d`             | `nn.Conv2d`                             |
+| `ConvTranspose1d`    | `nn.ConvTranspose1d`                    |
+| `ConvTranspose2d`    | `nn.ConvTranspose2d`                    |
+| `Dropout`            | `nn.Dropout` or `nn.Dropout2d` etc.     |
+| `Embedding`          | `nn.Embedding`                          |
+| `GateController`     | _No direct equivalent_                  |
+| `GELU`               | `nn.GELU`                               |
+| `Gru`                | `nn.GRU`                                |
+| `LayerNorm`          | `nn.LayerNorm`                          |
+| `Linear`             | `nn.Linear`                             |
+| `Lstm`               | `nn.LSTM`                               |
+| `MaxPool1d`          | `nn.MaxPool1d`                          |
+| `MaxPool2d`          | `nn.MaxPool2d`                          |
+| `MultiHeadAttention` | `nn.MultiheadAttention`                 |
+| `PositionalEncoding` | _No direct equivalent_                  |
+| `TransformerDecoder` | `nn.TransformerDecoder`                 |
+| `TransformerEncoder` | `nn.TransformerEncoder`                 |
```

Comment:
I would remove "a few" since it isn't that small and it will only grow, potentially making this line look odd in the future.

Also, the list isn't complete:

- CrossEntropyLoss
- MSELoss
- LayerNorm
- Embedding
- PositionalEncoding
- Relu

I would also separate them into sub-sections:

- Pooling
- Convolutions
- RNNs
- Transformer
- General
- Loss

At some point, we might create issues that document which modules we want to support in each section.

Finally, the `MSELoss` and `CrossEntropyLoss` don't implement `Module` yet, but we should annotate them even though they don't have state. They were always used inline, but we could put them into a module as is often done with PyTorch.

Reply:
> Finally, the `MSELoss` and `CrossEntropyLoss` don't implement `Module` yet, but we should annotate them even though they don't have state. They were always used inline, but we could put them into a module as is often done with PyTorch.

Filed a ticket for this: https://github.com/burn-rs/burn/issues/736

Rest is updated.